p11-kit/rpc-message.c

```diff
@@ -241,14 +241,25 @@ p11_rpc_message_write_attribute_buffer (p11_rpc_message *msg,
 	/* Write the number of items */
 	p11_rpc_buffer_add_uint32 (msg->output, num);
 
+	/* Is arr not null ? */
+	p11_rpc_buffer_add_byte (msg->output, arr ? 1 : 0);
+
+	/* force at least one attribute entry to be written since arr is not NULL */
+	if ((num == 0) &&
+	    (arr))
+		num = 1;
+
 	for (i = 0; i < num; ++i) {
 		attr = &(arr[i]);
 
 		/* The attribute type */
 		p11_rpc_buffer_add_uint32 (msg->output, attr->type);
 
+		/* is the value buffer not NULL ? */
+		p11_rpc_buffer_add_byte (msg->output, attr->pValue ? 1 : 0);
```

Comment:
The condition seems to be reverted. Here we add 1 if `attr->pValue` is *non*-NULL, but below you check the value with a variable `is_value_null`.

Reply:
the code is correct, but the comment is miss-misleading. I have aligned all the *_is_null so the follow the same semantic.

update will come in a minute.